packages/server/graphql/intranetSchema/mutations/loginSAML.ts

```diff
@@ -0,0 +1,108 @@
+import * as validator from '@authenio/samlify-node-xmllint'
+import base64url from 'base64url'
+import {GraphQLNonNull, GraphQLString} from 'graphql'
+import getSSODomainFromEmail from 'parabol-client/utils/getSSODomainFromEmail'
+import querystring from 'querystring'
+import * as samlify from 'samlify'
+import shortid from 'shortid'
+import getRethink from '../../../database/rethinkDriver'
+import AuthToken from '../../../database/types/AuthToken'
+import User from '../../../database/types/User'
+import encodeAuthToken from '../../../utils/encodeAuthToken'
+import {sendSegmentIdentify} from '../../../utils/sendSegmentEvent'
+import bootstrapNewUser from '../../mutations/helpers/bootstrapNewUser'
+import SAML from '../../../database/types/SAML'
+import {SSORelayState} from '../../queries/SAMLIdP'
+import LoginSAMLPayload from '../types/LoginSAMLPayload'
+
+const serviceProvider = samlify.ServiceProvider({})
+samlify.setSchemaValidator(validator)
+
+const getRelayState = (body: any) => {
+  const {RelayState} = body
+  let relayState = {} as SSORelayState
+  try {
+    relayState = JSON.parse(base64url.decode(RelayState))
+  } catch (e) {
+    // ignore
+  }
+  return relayState
+}
+
+const loginSAML = {
```

Comment:
SAML logic was really split between the endpoint & graphql, moved as much logic as possible inside graphql so we can easily draw the line between our web service & our graphql/processing service.